internal/controller/pkg/revision/deployment.go

```diff
@@ -171,52 +184,31 @@ func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRe
 	}
 	if revision.GetTLSServerSecretName() != nil {
 		mountTLSSecret(*revision.GetTLSServerSecretName(), tlsServerCertsVolumeName, tlsServerCertsDir, tlsServerCertDirEnvVar, d)
+		d.Spec.Template.Spec.Containers[0].Ports = append(d.Spec.Template.Spec.Containers[0].Ports,
+			corev1.ContainerPort{
+				Name:          webhookPortName,
+				ContainerPort: servicePort,
+			})
+		// for backward compatibility with existing providers, we set the
+		// environment variable WEBHOOK_TLS_CERT_DIR to the same value as
+		// TLS_SERVER_CERTS_DIR to ease the transition to the new certificates.
```

Comment:
What changes are required for providers to switch to the new structure? I assume it is just using `TLS_SERVER_CERTS_DIR` instead of `WEBHOOK_TLS_CERT_DIR` and `TLS_CLIENT_CERTS_DIR` instead of `ESS_TLS_CERTS_DIR`, anything else?
It could be helpful to open an example PR in one of the providers to demonstrate that.